docs/examples/te_llama/tutorial_accelerate_hf_llama_with_te.ipynb

```diff
@@ -0,0 +1,968 @@
+    "</figure>\n",
+    "\n",
+    "##### Self_Attn Layer\n",
+    "For simplicity in the block diagram illustration of the \"self_attn\" box, we omit the \"Grouped Query Attention\" operation and only showcase the modules which have associated weights.\n",
+    "   \n",
+    "##### MLP Layer\n",
+    "\n",
+    "SwiGLU is an activation defined as follows in the [modeling_llama.py](https://github.com/huggingface/transformers/blob/7c4995f93d8d24aae05e1e43279c96dce736e5c8/src/transformers/models/llama/modeling_llama.py#L236) file in the Hugging Face github repo:\n",
+    "```\n",
+    "down_proj = self.down_proj(self.act_fn(self.gate_proj(x)) * self.up_proj(x))\n",
+    "```\n",
+    "It requires a set of 3 weights as compared to 2 weights in conventional \"MLP\" layers e.g. in the traditional transformer architecture."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2a20adf8",
+   "metadata": {},
+   "source": [
+    "## [Baseline] Running HF `LlamaModel` (Precision: `BF16`)\n",
+    "\n",
+    "Llama2 weights are loaded into the Hugging Face native implementation `LlamaForCausalLM` (refer to [modeling_llama.py](https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py)). \n",
+    "\n",
+    "`batch_size` is `8` and precision is `BF16`\n",
+    "\n",
+    "The `LlamaDecoderLayer` is left unchanged in the baseline as follows:\n",
+    "\n",
+    "<figure align=\"center\">\n",
+    "<img src=\"media/llamadecoderlayer.png\" width=\"30%\">\n",
+    "    <figcaption> Fig 5: Revisiting \"LlamaDecoderLayer\". </figcaption>\n",
+    "</figure>\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "f794f2dc",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import torch\n",
+    "torch.set_warn_always(False)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "4ba1e069",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/usr/local/lib/python3.10/dist-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
+      "  from .autonotebook import tqdm as notebook_tqdm\n",
+      "You are attempting to use Flash Attention 2.0 with a model not initialized on GPU. Make sure to move the model to GPU after initializing it on CPU with `model.to('cuda')`.\n",
+      "Loading checkpoint shards: 100%|██████████| 2/2 [00:04<00:00,  2.25s/it]\n",
+      "Map: 100%|██████████| 9846/9846 [00:00<00:00, 12152.20 examples/s]\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "10 finetuning steps complete!\n",
+      "Average time taken per step: 289 milliseconds\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Restart the notebook (to flush the GPU memory)\n",
+    "from utils import restart_jupyter_notebook\n",
```

Comment:
You should add in the text that you will be using this file (and post a link to it)

Reply:
I've added a "note" now before the cell in which baseline is run. Does that help?